src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -58,12 +58,17 @@
 /**
  *
  */
-public class FsTranslog extends AbstractIndexShardComponent implements Translog {
+public class FsTranslog extends AbstractIndexShardComponent implements Translog, Closeable {
 
     public static final String INDEX_TRANSLOG_FS_TYPE = "index.translog.fs.type";
-    private static final String TRANSLOG_FILE_PREFIX = "translog-";
+    public static final String TRANSLOG_FILE_PREFIX = "translog-";
     private static final Pattern PARSE_ID_PATTERN = Pattern.compile(TRANSLOG_FILE_PREFIX + "(\\d+).*");
 
+
+    // this is a copy on write list and is in protected by any of the locks. The main reason
+    // is that is being accessed by two separate classes (additions & reading are done by FsTranslog, remove by FsView when closed)
+    private final List<FsView> outstandingViews = new CopyOnWriteArrayList<>();
```

Comment:
I think this should be a concurrentSet to be honest we remove stuff from this under locks and I don't like the runtime otherwise.